packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_table_v2.dart

```diff
@@ -0,0 +1,309 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:devtools_app_shared/ui.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+import '../../../shared/common_widgets.dart';
+import 'logging_model.dart';
+import 'logging_table_row.dart';
+
+/// A builder that includes an Offset to draw the context menu at.
+typedef ContextMenuBuilder = Widget Function(
+  BuildContext context,
+  Offset offset,
+);
+
+/// A Widget for displaying logs with line wrapping, along with log metadata.
+class LoggingTableV2 extends StatefulWidget {
+  const LoggingTableV2({super.key, required this.model});
+
+  final LoggingTableModel model;
+
+  @override
+  State<LoggingTableV2> createState() => _LoggingTableV2State();
+}
+
+class _LoggingTableV2State extends State<LoggingTableV2> {
```

Comment:
can you add a TODO to use `SearchControllerMixin` and `FilterControllerMixin` for search and filter functionality in this table?